crates/wasmparser/src/validator/operators.rs

```diff
@@ -1416,6 +1428,65 @@ where
         }
     }
 
+    fn cont_type_at(&self, key: TypeKey) -> Result<&ContType> {
+        let sub_ty = match key {
+            Either::A(at) => self.sub_type_at(at)?,
+            Either::B(id) => match self.resources.sub_type_at_id(id) {
+                None => bail!(self.offset, "unknown type {}", TypeIdentifier::index(&id)),
```

Comment:
Here we won't want to surface `TypeIdentifier::index` since that's just a purely internal index during validation. In these cases it's ok to jus say "unknown type" or "non-continuation type" or something like that without mentioning the index.